test/extended/openstack/volumes.go

```diff
@@ -241,6 +241,78 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack pla
 			o.Expect(err).NotTo(o.HaveOccurred())
 			o.Expect(out).To(o.Equal(fileContent))
 		})
+
+		g.It("should create a cinder volume when using cinder default storage class", func(ctx g.SpecContext) {
+
+			var err error
+
+			cinderSc := FindDefaultStorageClassByProvider(oc, "cinder.csi.openstack.org")
+			o.Expect(cinderSc).NotTo(o.BeNil(), "default cinder-csi storageClass not found.")
+
+			volumeClient, err = client.GetServiceClient(ctx, openstack.NewBlockStorageV3)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ns := oc.Namespace()
+			pvc := CreatePVC(ctx, clientSet, "cinder-pvc", ns, cinderSc.Name, "1Gi")
+			fileContent := "hello"
+
+			// Creating a deployment with the volumes attached
+			g.By("Creating Openshift deployment with 1 replica and cinder volume attached")
+			labels := map[string]string{"app": "cinder-test-dep"}
+
+			testDeployment := createTestDeployment(deploymentOpts{
+				Name:     "cinder-test-dep",
+				Labels:   labels,
+				Replicas: 1,
+				Protocol: v1.ProtocolTCP,
+				Port:     8080,
+				Volumes: []volumeOption{{
+					Name:      "data-volume",
+					PvcName:   pvc.Name,
+					MountPath: "data",
+				}},
+			})
+
+			deployment, err := clientSet.AppsV1().Deployments(ns).Create(ctx,
+				testDeployment, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			pods, err := oc.AdminKubeClient().CoreV1().Pods(ns).List(ctx, metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			pvcs, err := GetPvcsFromNamespace(ctx, dc, ns)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(pvcs)).To(o.Equal(1))
+
+			cinderVolumes, err := getVolumesFromName(ctx, volumeClient, pvcs[0].Get("spec.volumeName").String())
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(cinderVolumes)).To(o.Equal(1))
+			volumeID := cinderVolumes[0].ID
+			_, err = volumes.Get(ctx, volumeClient, volumeID).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Create a file in the pod's attached volume, restart it and read what was written")
+			_, err = oc.Run("exec").Args(pods.Items[0].Name, "--", "bash", "-c", fmt.Sprintf("echo %s > /data/1", fileContent)).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Delete pod so it get restarted")
+			err = oc.Run("delete").Args("pod", pods.Items[0].Name).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			_, err = exutil.WaitForPods(clientSet.CoreV1().Pods(ns),
+				exutil.ParseLabelsOrDie(""), exutil.CheckPodIsRunning, 1, 3*time.Minute)
```

Comment:
Although not crucial in this case I wonder if not to use CheckPodIsReady

Reply:
Ack